api/host_authentication.py

```diff
@@ -1,8 +1,18 @@
+import base64
+import hmac
+import hashlib
+import os
+
 from django.contrib.auth.models import User
 from rest_framework import authentication
 from rest_framework import exceptions
 
 
+def generate_woocommerce_signature(body, secret):
```

Comment:
is there a particular reason why this is not encapsulated within `SignatureValidation`? Is this the Python way?